frame/support/src/lib.rs

```diff
@@ -93,6 +92,72 @@ pub mod unsigned {
 	};
 }
 
+pub mod weights {
+	use crate::dispatch;
+	pub use sp_weights::*;
+
+	#[deprecated = "Function has moved to `frame_support::dispatch`"]
+	pub fn extract_actual_pays_fee(
+		res: &dispatch::DispatchResultWithPostInfo,
+		info: &dispatch::DispatchInfo,
+	) -> dispatch::Pays {
+		dispatch::extract_actual_pays_fee(res, info)
+	}
+	#[deprecated = "Function has moved to `frame_support::dispatch`"]
+	pub fn extract_actual_weight(
+		res: &dispatch::DispatchResultWithPostInfo,
+		info: &dispatch::DispatchInfo,
+	) -> Weight {
+		dispatch::extract_actual_weight(res, info)
+	}
+	#[deprecated = "Trait has moved to `frame_support::dispatch`"]
+	pub trait ClassifyDispatch<T>: dispatch::ClassifyDispatch<T> {
+		fn classify_dispatch(&self, target: T) -> dispatch::DispatchClass {
+			<Self as dispatch::ClassifyDispatch<T>>::classify_dispatch(self, target)
+		}
+	}
+	#[deprecated = "Enum has moved to `frame_support::dispatch`"]
+	pub type DispatchClass = dispatch::DispatchClass;
+	#[deprecated = "Struct has moved to `frame_support::dispatch`"]
+	pub type DispatchInfo = dispatch::DispatchInfo;
+	#[deprecated = "Trait has moved to `frame_support::dispatch`"]
+	pub trait GetDispatchInfo: dispatch::GetDispatchInfo {
+		fn get_dispatch_info(&self) -> dispatch::DispatchInfo {
+			<Self as dispatch::GetDispatchInfo>::get_dispatch_info(self)
+		}
+	}
+	#[deprecated = "Trait has moved to `frame_support::dispatch`"]
+	pub trait OneOrMany<T>: dispatch::OneOrMany<T> {
+		fn into_iter(self) -> Self::Iter where Self: Sized {
+			<Self as dispatch::OneOrMany<T>>::into_iter(self)
+		}
+	}
+	#[deprecated = "Enum has moved to `frame_support::dispatch`"]
+	pub type Pays = dispatch::Pays;
+	#[deprecated = "Trait has moved to `frame_support::dispatch`"]
+	pub trait PaysFee<T>: dispatch::PaysFee<T> {
+		fn pays_fee(&self, target: T) -> dispatch::Pays {
+			<Self as dispatch::PaysFee<T>>::pays_fee(self, target)
+		}
+	}
+	#[deprecated = "Struct has moved to `frame_support::dispatch`"]
+	pub type PerDispatchClass<T> = dispatch::PerDispatchClass<T>;
+	#[deprecated = "Struct has moved to `frame_support::dispatch`"]
+	pub type PostDispatchInfo = dispatch::PostDispatchInfo;
+	#[deprecated = "Trait has moved to `frame_support::dispatch`"]
+	pub trait WeighData<T>: dispatch::WeighData<T> {
+		fn weigh_data(&self, target: T) -> Weight {
+			<Self as dispatch::WeighData<T>>::weigh_data(self, target)
+		}
+	}
+	#[deprecated = "Trait has moved to `frame_support::dispatch`"]
+	pub trait WithPostDispatchInfo: dispatch::WithPostDispatchInfo {
+		fn with_weight(self, actual_weight: Weight) -> dispatch::DispatchErrorWithPostInfo where Self: Sized {
+			<Self as dispatch::WithPostDispatchInfo>::with_weight(self, actual_weight)
+		}
+	}
```

Comment:
I'm attempting to try a novel way to deprecate types that have been moved, and one of the ways that I can think of is to create new subtraits and type alias and immediately deprecating them, telling the FRAME dev that the trait/type has been moved to another location.

Would be nice if I get some eye balls on this to ensure that this is worthwhile to pursue.

Reply:
The idea here is to create code that still works like the old way, but would emit a deprecation warning about the usage. This would make it easier for people to migrate over to the new location by allowing them to do so at a time of their choosing, while still having code that works.

Reply:
looks decent to me, just not sure when and based on what criteria are we going to remove the old code? we also have not removed the old macros now for more than a year and it is unclear to me when we are going to. 

If there's no clear answer to this, I think it is also reasonable to just break the code downstream instead of keeping a lot of duplicate code around.

Reply:
You could stick a comment stating that the old methods will be removed once we get to date x.